src/Components/Component.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
```

Comment:
Please use #define guard instead of pragma once. See:
https://google.github.io/styleguide/cppguide.html#The__define_Guard